pkg/awsutils/awsutils.go

```diff
@@ -740,11 +740,16 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 
 // AllocENI creates an ENI and attaches it to the instance
 // returns: newly created ENI ID
-func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string) (string, error) {
-	eniID, err := cache.createENI(useCustomCfg, sg, subnet)
+func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string, numIPs int) (string, error) {
+	eniID, err := cache.createENI(useCustomCfg, sg, subnet, numIPs)
 	if err != nil {
 		return "", errors.Wrap(err, "AllocENI: failed to create ENI")
 	}
+	// This is to handle the case where we need 0 IPs, so ENI will not be created
+	if eniID == "" {
```

Comment:
Why are we even entering AllocENI the num of IPs is 0?

Reply:
Refactoring based on out offline discussion (Don't call AllocENI if we need 0 IPs)